ComSemApp/templates/ComSemApp/student/assessment_results.html

```diff
@@ -80,10 +80,91 @@ <h1 class="card-title">Review complete!</h1>
     </div>
 </div>
+<!-- Modal -->
+<div class="modal fade" id="confirm-modal" tabindex="-1" role="dialog" aria-labelledby="confirm-modal-label" aria-hidden="true">
+    <div class="modal-dialog" role="document">
+        <div class="modal-content">
+            <div class="modal-header">
+                <h5 class="modal-title" id="confirm-modal-label">Are you sure?</h5>
+                <button type="button" class="close" data-dismiss="modal" aria-label="Close">
+                <span aria-hidden="true">&times;</span>
+                </button>
+            </div>
+            <div class="modal-body">
+                <span id="confirm-message"><b>Are you sure you want to request teacher review?</b><br>
+                    Please listen back to your audio recording to ensure that there was a grading or 
+                    transcription error before requesting review.<br>
+                    If you have already done so, continue. 
+                </span>
+            </div>
+            <div class="modal-footer">
+                <button type="button" class="btn btn-secondary" data-dismiss="modal">Cancel</button>
+                <button type="button" class="btn btn-warning" id="confirm-button">Request Review</button>
+            </div>
+        </div>
+    </div>
+</div>
+  
+  
 <div class="btn-group d-flex card-footer" role="group" aria-label="...">
     <a href="{% url 'student:course' course.id %}" class ="btn-success btn w-100 m-1">Return Home</a>
     <a href="{% url 'student:course' course.id %}" class ="btn-success btn w-100 m-1">Try Again</a>
 </div>
+
 <script src="{% static "ComSemApp/js/ComSemRecording-opus.js" %}?{% now "U" %}"></script>
+<script type="text/javascript">
+
+// Add click event listener to all request buttons
+$(".request-btn").on("click", (obj) => {
+    // Get the request ID from the data attribute of the button
+    // obj.currentTarget is needed because obj holds information regarding the event
+    // currentTarget grabs the item which initaited the event
+    // this is similar to just using the 'this' call but within an arrow annonymous function
+    var attempt_id = $(obj.currentTarget).data("submit-id");
+  
+    // Show the modal dialog box
+    $("#confirm-modal").modal("show");
+  
+    // Add click event listener to the confirm button
+    $("#confirm-button").one("click", (obj) => {
+      // Submit the form with the submit ID as a parameter
+      console.log("confirm clicked")
+        teacherRequestClick(attempt_id);
+      // Hide the modal dialog box
+      $("#confirm-modal").modal("hide");
+    });
+  });
+
+function teacherRequestClick(attempt_id){
+    if (!$("#request" + attempt_id).hasClass("disabled")){
+        console.log(attempt_id);
+        $("#request" + attempt_id).removeClass("btn-warning").addClass("btn-secondary").prop("disabled", true);
+
+        let requestFormData = new FormData(); // creates new form
+        requestFormData.append( 'attempt_id', attempt_id);
```

Comment:
To reiterate what I said above, this form data should be labeled `'attempt'` so the backend can enforce constraints automatically